### PR TITLE
LU to Factorization tests

### PR DESCRIPTION
**Pull request by kshyatt**
Just making sure that the eltype conversion one works.

### CODE REVIEW

File: stdlib/LinearAlgebra/test/lu.jl [hunk 1]
```diff
@@ -72,6 +72,11 @@ dimg  = randn(n)/2
             bft = eltya <: Real ? LinearAlgebra.LU{BigFloat} : LinearAlgebra.LU{Complex{BigFloat}}
             bflua = convert(bft, lua)
             @test bflua.L*bflua.U ≈ big.(a)[p,:] rtol=ε
+            @test Factorization{eltya}(lua) == lua
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/lu.jl]: If this is meant to be a no-op in the case of matching eltypes, maybe strengthen this to `===` comparison?

File: stdlib/LinearAlgebra/test/lu.jl [hunk 2]
```diff
@@ -72,6 +72,11 @@ dimg  = randn(n)/2
...
+            # test Factorization with different eltype
+            if eltya <: BlasReal
+                @test Array(Factorization{Float16}(lua)) ≈ Array(lu(convert(Matrix{Float16}, a)))
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/lu.jl]: What exactly do you want to test here? The l.h.s. is even approx to the original `a`, so somehow you're not yet capturing that the conversion of the factorization was effective. Maybe (also) test the eltype of some factor or something like that?

Comment (kshyatt) [reply stdlib/LinearAlgebra/test/lu.jl]: I think this question is only answerable by my highly sleep deprived Thursday self... I'll update :)

Comment (kshyatt) [reply stdlib/LinearAlgebra/test/lu.jl]: OK, tried with `==` and the test failed due to what appear to be rounding issues.

**Pull request merged**